Add a perf test for CSV map import

kalman opened the pull request:
Currently we only have a perf test for CSV list import, which uses the
sf-crime dataset. This test uses the 43MB sf-registered-businesses
dataset instead, since sf-crime is too slow. Which is ironic, since we
normally parse sf-crime into a map.

I've also tightened up some of the other perf tests.
- Pause the blob import tests during commit, because commit is much
  faster the first time. I'll be able to reduce the test repetitions.
- Make the pure CSV parsing test use a smaller dataset, it doesn't need
  to use something as large as ny-vehicle-registrations.

Commit:
Add a perf test for CSV map import

Currently we only have a perf test for CSV list import, which uses the
sf-crime dataset. This test uses the 43MB sf-registered-businesses
dataset instead, since sf-crime is too slow. Which is ironic, since we
normally parse sf-crime into a map.

I've also tightened up some of the other perf tests.

- Pause the blob import tests during commit, because commit is much
  faster the first time. I'll be able to reduce the test repetitions.
- Make the pure CSV parsing test use a smaller dataset, it doesn't need
  to use something as large as ny-vehicle-registrations.

kalman commented:
Perf this

kalman commented on samples/go/csv/csv-import/perf_test.go:
```diff
@@ -50,29 +46,57 @@ func (s *perfSuite) Test01ImportSfCrimeBlobFromTestdata() {
 	defer s.closeGlob(raw)
```
I might rename this test as well, like `Test01ImportSfCrimeBlobOmitCommit`, so that the test results don't appear to change.

rafael-atticlabs commented on samples/go/csv/csv-import/perf_test.go:
```diff
@@ -50,29 +46,57 @@ func (s *perfSuite) Test01ImportSfCrimeBlobFromTestdata() {
-	_, err := ds.CommitValue(blob)
-	assert.NoError(err)
+	// CommitValue is much faster the first time, so pause the test while it's happening.
+	s.Pause(func() {
```
per meatspace: this seems a bit strange to me. it might be better to leave this time in and then have a way in the UI to show "hot" vs "cold" perf.

rafael-atticlabs commented:
lgtm. "hot" vs "cold" suggestion doesn't need to be addressed here neccesarily

Commit:
Slightly different approach

This version of the patch achieves the same high-level goals, but
instead of attempting to account for the first run being faster, just
entirely re-creates the test on each run (there was a bug where the
database was per suite, not per run).

I also just removed all I/O ambiguity by reading files into a buffer.

Commit:
Un-comment code

kalman commented:
Sorry @rafael-atticlabs , PTAL at the 2nd patch. It's changed non-trivially. Almost enough to warrant a new patch, but it's all to achieve the same goal anyway, so it makes sense to be here.

kalman commented:
Perf this

rafael-atticlabs commented on samples/go/csv/csv-import/perf_test.go:
```diff
@@ -85,32 +98,26 @@ func (s *perfSuite) TestParseNyVehicleRegistrations() {
 
 	s.Pause(func() {
-		glob, err := filepath.Glob(pattern)
+		glob, err := filepath.Glob(path.Join(pattern...))
```
mild pref for not excluding the input io time.

rafael-atticlabs commented:
lgtm

Commit:
Don't read to buffer nor Pause

Commit:
Cleanup

kalman commented:
FYI no longer pausing nor using buffer
